app/models/user_summary.rb

```diff
@@ -145,7 +145,7 @@ def top_categories
 
     top_categories = {}
 
-    Category.where(id: post_count_query.limit(MAX_SUMMARY_RESULTS).pluck('category_id'))
+    Category.where(id: post_count_query.order("count(*) DESC").limit(MAX_SUMMARY_RESULTS).pluck('category_id'))
```

Comment:
Isn't `count(*)` potentially quite slow on psql? The post table is quite big.

Reply:
`post_count_query` is limited to posts from a specific user, so it's not the entire table. We are already doing a number of `.pluck("count(*)")` further down in this function, so I don't think adding it here is going to cause too much issue.

But you are right - it's probably not ideal for us to be doing these massive queries for every user profile view. The good news is that @xfalcox added some caching a while back, so it doesn't happen for **every** view...

I think to make it more efficient we'd need to set up a `user_category_stats` table with a cache of post/topic count for each category/user pair. Possible, but it would be quite a large refactor.

Reply:
Ok, sounds good.